### DISCUSSION TITLE
Restructure Supervision Library (functions,imports,classes,utils etc)

### DISCUSSION THREAD

**Discussion by onuralpszr** [roboflow/supervision#306] [Ideas]
Hello everyone 👋 

I wanted to take a moment to share an idea that I believe could significantly improve the structure and usability of our Supervision Library. After spending some time working with the current setup, I've identified a few areas where reorganizing functions, classes, imports, and utilities could lead to a more intuitive and efficient experience for users. Allow me to outline the proposed changes in a clearer format; 

**Function Repositioning**: Some of the core functions could be more logically positioned to improve accessibility. For example, moving merge function in Detection file from its current location to maybe more utility place. Or Split detection types into separate files so we can reduce the size of "core.py" and makes code more readable 

**Class Refactoring**: Certain classes might benefit from a new naming structure to reflect their roles more accurately. 

**Import Optimization**: Review the import statements across the library to ensure they are grouped logically and avoid unnecessary redundancy. This can simplify code maintenance and enhance readability.

**Utility Directory**: Create a dedicated directory for utility modules that are frequently used across the library. This central repository would prevent duplication and provide a single source of truth for common utility functions.

Of course those also require document before work and if require draw diagram for proper position to move certain pieces into new place and try to keep ABI as same as possible but eventually deprecate old usages and remove them completely after releases. Supervision did a very strong and fast start and growing bigger. My first initiative was moving to poetry and then move into pre-commit to give better standard and automation, adding more feature require sometimes require move house cleaning/keeping, so for that reason I invite you to all share your thoughts and which you think it should be change or what would be better option so we can discuss and collect to ideas and decide what will do next. As much as I like work on Supervision, we want to keep shiny as well :) I hope you all have great day and thank you in advance for your ideas.

**Comment by hardikdava** [comment]
@onuralpszr , I would say this is needed for longer run. I agree with you.

**Reply by SkalskiP** [reply]
Yeah, I agree. We spoke about it with @hardikdava some time ago. @onuralpszr, like we agreed during our call , let's define the scope of the changes and actions we want to take.
